cassandra/rab.py

```diff
@@ -0,0 +1,220 @@
+"""Definition of the RAB class.
+
+The RAB provides access to capabilities hosted on remote nodes in MP
+calculations.  It implements much of the component interface, but it isn't a
+component as such, and in some respects it works very differently to proper
+components.
+
+The name RAB was originally an acronym for "Remote Access Broker", but mostly
+it's just a name.  Also, "RAB" are my mom's initials.  Hi, Mom!
+
+This module should _only_ be imported by the my.py module.  It imports mpi4py,
+which will have the side effect of trying to initialize MPI if mp.py hasn't
+already done it.  
+"""
+
+from cassandra.constants import TAG_REQ, TAG_REQID_BASE
+from mpi4py import MPI
+import concurrent.futures as ft
+import threading
+import logging
+
+# The choice of the sleep length in RAB listener loop is a tradeoff.  Making it
+# shorter improves responsiveness, but creates more waiting in the GIL,
+# potentially slowing down threads that are actually running models (especially
+# if those models are implemented in python and running in the same process).  
+RAB_LOOP_SLEEP = 0.05          # 50 ms
+
+class RAB(object):
```

Comment:
Inheriting from `object` is the default in python 3, so this could just be `class RAB:`.

Reply:
Good to know, thanks.